js/battle.js

```diff
@@ -4010,9 +4019,10 @@ var Battle = (function () {
 				var ability = Tools.getAbility(args[2]);
 				var effect = Tools.getEffect(kwargs.from);
 				var ofpoke = this.getPokemon(kwargs.of);
-				poke.ability = ability.name;
 				if (!effect.id || kwargs.fail) {
```

Comment:
Perhaps this part should also be implemented in `markAbility`? Perhaps as a second argument.